pylint_nautobot/q_search_filter.py

```diff
@@ -0,0 +1,39 @@
+"""Check for use of SearchFilter on NautobotFilterSets instead of custom `q` search functions."""
+
+from astroid import Assign, AssignName, ClassDef, FunctionDef
+from pylint.checkers import BaseChecker
+
+from pylint_nautobot.constants import MSGS
+
+
+class NautobotUseSearchFilterChecker(BaseChecker):
+    """Visit NautobotFilterSet subclasses and check for use of `q = SearchFilter`, instead of `q = django_filters.CharField`."""
+
+    version_specifier = ">=2,<3"
+
+    name = "nautobot-use-search-filter"
+    msgs = {
+        **MSGS.C4272,
+        **MSGS.C4273,
+        **MSGS.C4274,
+    }
+
+    def visit_classdef(self, node: ClassDef):
+        """Visit class definitions."""
+        for ancestor in node.ancestors():
+            if ancestor.qname() != "nautobot.extras.filters.NautobotFilterSet":
+                return
+
+            for child_node in node.get_children():
+                if isinstance(child_node, Assign) and any(
+                    isinstance(target, AssignName) and target.name == "q" for target in child_node.targets
+                ):
+                    child_node_name = child_node.value.func.as_string().split(".")[-1]
+                    # The `q` attribute should not be a CharFilter, instead it should be a SearchFilter
+                    if child_node_name == "CharFilter":
+                        self.add_message("nb-no-char-filter-q", node=child_node)
```

Comment:
Any straightforward way to replace strings like `"nb-no-char-filter-q"` with a constant as well so as to keep these in sync with `MSGS`?

Reply:
I think so, I will change this as well.

Reply:
No, not straight forward after testing.